app/src/main/java/com/example/solar_alarm/CreateAlarm/CreateAlarmFragment.java

```diff
@@ -209,43 +187,117 @@ public void onClick(View v) {
         return view;
     }
 
-    private void scheduleAlarm() {
-        int alarmId = new Random().nextInt(Integer.MAX_VALUE);
-
-        Alarm alarm = new Alarm(
-                alarmId,
-                TimePickerUtil.getTimePickerHour(timePicker),
-                TimePickerUtil.getTimePickerMinute(timePicker),
-                title.getText().toString(),
-                System.currentTimeMillis(),
-                true,
-                recurring.isChecked(),
-                mon.isChecked(),
-                tue.isChecked(),
-                wed.isChecked(),
-                thu.isChecked(),
-                fri.isChecked(),
-                sat.isChecked(),
-                sun.isChecked()
-        );
-
-        createAlarmViewModel.insert(alarm);
-
-        alarm.schedule(getContext());
+    @RequiresApi(api = Build.VERSION_CODES.O)
+    public SolarTime getSolarTime(Location locationItem, Calendar date) throws ExecutionException, InterruptedException {
+        boolean isLocationIdDatePairExists = getLocationIdDatePareExists(locationItem, date);
+        SolarTime solarTimeItem = new SolarTime();
+
+        if(!isLocationIdDatePairExists) {
+            SunriseSunsetRequest sunriseSunsetRequest = new SunriseSunsetRequest((float) locationItem.Latitude, (float) locationItem.Longitude, date, true);
+            try {
+                solarTimeItem = new TimeResponseTask().execute(sunriseSunsetRequest, locationItem).get();
+                timeZoneConverter = new TimeZoneConverter(solarTimeItem);
+                solarTimeItem = timeZoneConverter.convertSolarTime();
+                sunriseData.setText(solarTimeItem.Sunrise.toString());
+                solarNoonData.setText(solarTimeItem.SolarNoon.toString());
+                sunsetData.setText(solarTimeItem.Sunset.toString());
+                solarTimeRepository.Insert(solarTimeItem);
+
+            } catch (ExecutionException e) {
+                e.printStackTrace();
```

Comment:
Fix all the swallowed exceptions. `throw e` so you get a crash at the point of the exception.